tests/test_generics.py

```diff
@@ -1071,3 +1071,82 @@ class GModel(GenericModel, Generic[FieldType, ValueType]):
     Fields = Literal['foo', 'bar']
     m = GModel[Fields, str](field={'foo': 'x'})
     assert m.dict() == {'field': {'foo': 'x'}}
+
+
+@skip_36
+def test_generic_subclass():
+    T = TypeVar('T')
+
+    class A(GenericModel, Generic[T]):
+        ...
+
+    class B(A[T], Generic[T]):
+        ...
+
+    assert B[int].__name__ == 'B[int]'
+    assert issubclass(B[int], B)
+    assert issubclass(B[int], A[int])
```

Comment:
Probably should add a few negative cases to each of these tests. For example

    assert not issubclass(B[str], A[int])

Reply:
Good point, I've added some suitable negative cases.